core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServices.java

```diff
@@ -97,7 +112,7 @@
      * Returns the list of {@link Path} annotated methods.
      */
     private static List<Method> requestMappingMethods(Object object) {
-        return Arrays.stream(object.getClass().getMethods())
+        return Arrays.stream(object.getClass().getDeclaredMethods())
```

Comment:
Need to climb up the class hierarchy because `getDeclaredMethods()` excludes the inherited methods.